language/bazel/visibility/lang.go

```diff
@@ -64,7 +66,19 @@ func (*visibilityExtension) GenerateRules(args language.GenerateArgs) language.G
 	}
 
 	r := rule.NewRule("package", "")
-	r.SetAttr("default_visibility", cfg.visibilityTargets)
+	for _, er := range args.File.Rules {
+		if er.Kind() == "package" {
+			r = er
```

Comment:
I'm curious why this was needed. Extensions should generate new rules and return them through `res.Gen`, relying on Gazelle to merge generated and existing rules. If an extension modifies an existing rule directly, it can break `# keep` directives or cause other unexpected behavior.